File: jovy-backend/src/services/tag.service.ts
import { ITagService } from '../abstractions/interfaces';
import { BaseService } from '../core/base.service';
import { Tag } from '../models';
import { In } from 'typeorm';
import { TagDto } from '../abstractions/dtos';
import { updateEntityFields } from '../shared/utils';
import { UPDATE_TAG_FIELDS } from '../constants/common';

export class TagService extends BaseService<Tag> implements ITagService {
  async getByIds(ids: number[] = []): Promise<Tag[]> {
    return this.repository.findBy({ id: In(ids) });
  }

  async create(dto: Partial<TagDto>): Promise<Tag> {
    const { name, color } = dto;
    return this.repository.save(this.repository.create({ name, color }));
  }

  async change(id: number, dto: Partial<TagDto>): Promise<Tag> {
    const tag: Tag = await this.getOne(id);
    return this.repository.save(updateEntityFields(tag, dto, UPDATE_TAG_FIELDS));
  }
}

File: backend/src/main/kotlin/taskmanager/backend/services/TagService.kt
package taskmanager.backend.services

import org.bson.types.ObjectId
import taskmanager.backend.models.Tag
import taskmanager.backend.services.base.AttachedToProjectEntityService

interface TagService : AttachedToProjectEntityService<Tag> {
    suspend fun getByIds(ids: List<ObjectId>): List<Tag>
    suspend fun create(userId: ObjectId, projectId: ObjectId, name: String): Tag
    suspend fun updateById(id: ObjectId, name: String): Tag
}